src/lib/utils/analytics.ts

```diff
@@ -3,42 +3,73 @@
 import type { AnalyticsPayload, PageProps } from '../types/analytics';
 import crypto from 'node:crypto';
 
+declare global {
+	interface Window {
+		prevPages?: string[];
+	}
+}
+
 export const generateHash = async (value: string) => {
 	const hash = crypto.createHash('sha512').update(value).digest('hex');
 	return hash;
 };
 
+const getPageProps = () => {
+	return {
+		path: window.location.pathname,
+		url: window.location.href,
+		search: window.location.search,
+		title: document.title,
+		referrer: window.prevPages ? window.prevPages[window.prevPages.length - 1] : undefined
+	};
+};
+
 export const trackEvent = async (eventName: string, props: any) => {
+	const pageProps = getPageProps();
+
 	const body: AnalyticsPayload = {
 		type: 'event',
 		eventName,
-		props
+		props: props,
+		context: { page: pageProps }
 	};
+
+	console.log('Trying track call with payload:');
+	console.dir(body);
```

Comment:
There are a couple of logs. Can we get rid of them, or can we delete them before merging?